File: backend/utils/schema.py
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, validator


class MyBaseModel(BaseModel):
    @validator("*", pre=True)
    def empty_str_to_none(cls, v):
        if v == "":
            return None
        return v


class Plant(MyBaseModel):
    PlantID: str
    HumanName: Optional[str] = None
    Species: Optional[str] = None
    Location: str
    ParentID: Optional[str] = None
    Source: str
    SourceDate: str
    Sink: Optional[str] = None
    SinkDate: Optional[datetime] = None
    Notes: Optional[str] = None


class Image(MyBaseModel):
    ImageID: str
    PlantID: str
    S3Url: str
    Timestamp: datetime

File: plant-app/src/schema.ts
export interface PlantImage {
  ImageID: string;
  Timestamp: string;
  SignedUrl: string;
}

export interface Plant {
  plant_id: string;
  human_id: number;
  human_name: string;
  species: string | undefined;
  location: string;
  // TODO: figure out how to handle this as list of int
  parent_id: string[] | undefined;
  source: string;
  source_date: string;
  sink: string | undefined;
  sink_date: string | undefined;
  notes: string | undefined;
}
